File: 01_Language/01_Functions/js/array_fill_keys.js
function array_fill_keys(keys, value) {
    let res = {}
    keys.forEach(k => {
        res[k] = value
    })
    return res
}

let keys = ['foo', 5, 10, 'bar']
console.log(array_fill_keys(keys, 'banana'))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/ArrayFillKeys.java
package com.onevgo.functions;

import java.util.*;
import java.util.function.Function;
import java.util.stream.Collectors;

public class ArrayFillKeys {
    public static <K, V> Map<K, V> arrayFillKeys(Collection<K> keys, V value) {
        return keys.stream().collect(Collectors.toMap(Function.identity(), v -> value, (v, v2) -> v2, HashMap::new));
    }

    public static void main(String[] args) {
        List<Object> list = Arrays.asList("foo", 5, 10, "bar");
        System.out.println(arrayFillKeys(list, "banana"));
    }
}
